client.go

```diff
@@ -201,14 +201,16 @@ func (co *Conn) ReadMsg() (*Msg, error) {
 		}
 	}
 	n, err := co.Read(p)
-	if err != nil && n == 0 {
+	if err != nil {
 		return nil, err
+	} else if n < 12 {
```

Comment:
[ Quoting <notifications@github.com> in "Re: [dns] Added function for lazy m..." ]
>>  		return nil, err
>> +	} else if n < 12 {
>
>Any reason to choose private vs making it public?

Could as well be public.